### ISSUE TITLE
/x/net/context based routing

### ISSUE THREAD

**Issue by abrykajlo** [manyminds/api2go#230]
I have been working on an old fork of ours to convert your routing system to make using golang.org/x/net/context a lot easier for extending functionality of api2go with custom routes.

Only thing is it would introduce breaking changes on routes and middleware. The nice things is that it would allow any new routes to have access to whatever the middleware sets in the context.

Let me know what you think and if you would like something based of your master branch.

[here it is](https://github.com/picatic/api2go/blob/contextual/routing/router.go)

**Comment by wwwdata** [comment]
hi, yes that makes sense. Currently, the Context can only be reached in our CRUD functions, because you can get it out of the `Request` struct.

We just have to make sure that everything all the [api2go-adapters](https://github.com/manyminds/api2go-adapter) works with that too.

**Comment by abrykajlo** [comment]
So if I manage to get api2go-adapters working with my solution would you consider it a viable solution @wwwdata

**Comment by wwwdata** [comment]
yes. What do you think about it @sharpner ?

**Comment by sharpner** [comment]
yeah that would be awesome. Also I agree, it has to work with the routers. But BC breaks are not that problematic at the moment.

**Comment by abrykajlo** [comment]
Cool I will work on this then

**Comment by sharpner** [comment]
thank you! Much appreciated!

**Comment by arcijsg** [comment]
I'm really looking forward too for ability to define alternative routes and still benefit from the data that is available to api2go automatically - defined resource routes.

For example, to be able to POST to /order/11/line instead of (or additionally to) posting to /lines and adding order id as a relationship in the POST body)

**Comment by abrykajlo** [comment]
@arcijsg I am doing what I can for this. We are using api2go at work and it's seems like it might become a priority to get this done. Work has been busy.

**Comment by arcijsg** [comment]
@abrykajlo, did You have a chance to find a solution, or a different approach has emerged since?
